ui/app/routes/vault/cluster/clients/counts.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import timestamp from 'core/utils/timestamp';
+import { getUnixTime } from 'date-fns';
+
+import type StoreService from 'vault/services/store';
+import type { ClientsRouteModel } from '../clients';
+
+export default class ClientCountTokenRoute extends Route {
+  @service declare readonly store: StoreService;
+
+  queryParams = {
+    start_time: {
+      refreshModel: true,
+    },
+    end_time: {
+      refreshModel: true,
+    },
+  };
+
+  currentTimestamp = getUnixTime(timestamp.now());
+
+  async getActivity(start_time: number, end_time: number) {
+    let activity, activityError;
+    try {
+      activity = await this.store.queryRecord('clients/activity', {
+        start_time: { timestamp: start_time },
+        end_time: { timestamp: end_time },
+      });
+    } catch (error) {
+      activityError = error;
+    }
+    return [activity, activityError];
+  }
+
+  async model(params: { start_time: string; end_time: string }) {
+    const { config } = this.modelFor('vault.cluster.clients') as ClientsRouteModel;
+    const startTimestamp = Number(params.start_time) || getUnixTime(config.billingStartTimestamp);
```

Comment:
previously we returned null for `licenseStartTimestamp` so the user could input a date if we didn't have one to make the initial request with. It's possible a user may not have permissions for cluster config and so this value would be undefined.

How is this handled now?

Reply:
That may have deviated then. I think I mistakenly removed the condition to only query activity if there is a start time thinking that we should go ahead and then display the error. Let me add that back in. The template should still be setup to handle that case but I'll write a test to cover it as well.